### PR TITLE
Correct incomplete_array_typet [DOC-12]

### PR DESCRIPTION
**Pull request by majakusber**
This is part of the `typet` cleanup, based on top of https://github.com/diffblue/cbmc/pull/2815, please review only the last two commits.

### CODE REVIEW

**Comment by tautschnig** [comment]
Happy to approve once the depended-on commits are merged.

File: src/util/std_types.h
Comment (thk123) [inline src/util/std_types.h]: So this is a change in behaviour - have you checked that this constructor wasn't used anywhere? CLion says no, so you might just like to delete it?

**Review by thk123 — APPROVED** [review]
Assuming this passes CI and once preceding PR has been merged

File: src/util/std_types.h
Comment (majakusber) [reply src/util/std_types.h]: It seems like it's not used indeed, removed.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 6abec0527f479277feb94fdfb00e6e6a6cbdb598).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84464293

**Comment by majakusber** [comment]
diffblue/cbmc#2815 is merged now, rebased on develop.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 5660eeafdea65bd3b358f2c1a819e820ab460cf5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85007431
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: ff5a2320e127c2c01d22469f395f53757bd880b1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85009844

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**
